monkey/infection_monkey/island_api_client/http_client.py

```diff
@@ -59,23 +59,26 @@ def __init__(self, retries=RETRIES):
         self._session = requests.Session()
         retry_config = Retry(retries)
         self._session.mount("https://", HTTPAdapter(max_retries=retry_config))
-        self._api_url: Optional[str] = None
+        self._server_url: Optional[str] = None
+        self._additional_headers: Optional[Dict[str, Any]] = None
 
-    @handle_island_errors
-    def connect(self, island_server: SocketAddress):
-        try:
-            self._api_url = f"https://{island_server}/api"
-            # Don't use retries here, because we expect to not be able to connect.
-            response = requests.get(  # noqa: DUO123
-                f"{self._api_url}?action=is-up",
-                verify=False,
-                timeout=MEDIUM_REQUEST_TIMEOUT,
-            )
-            response.raise_for_status()
-        except Exception as err:
-            logger.debug(f"Connection to {island_server} failed: {err}")
-            self._api_url = None
-            raise err
+    @property
+    def server_url(self):
+        return self._server_url
+
+    @server_url.setter
+    def server_url(self, endpoint: Optional[str]):
+        if endpoint:
+            endpoint = f"https://{endpoint}"
+        self._server_url = endpoint
+
+    @property
+    def additional_headers(self):
+        return self._additional_headers
```

Comment:
Does anything outside `http_client` pull `additional_headers`? If not, we can get rid of this public getter.

Reply:
Couldn't update without a getter